src/test/java/teammates/test/cases/ui/pagedata/InstructorCourseStudentDetailsPageDataTest.java

```diff
@@ -0,0 +1,168 @@
+        assertEquals(pictureUrl, studentProfile.getPictureUrl());
+        assertEquals(inputStudent.name, studentProfile.getName());
+        assertEquals(inputStudentProfile.shortName, studentProfile.getShortName());
+        assertEquals(inputStudentProfile.gender, studentProfile.getGender());
+        assertEquals(inputStudentProfile.email, studentProfile.getEmail());
+        assertEquals(inputStudentProfile.institute, studentProfile.getInstitute());
+        assertEquals(inputStudentProfile.nationality, studentProfile.getNationality());
+        assertEquals(inputStudentProfile.moreInfo, studentProfile.getMoreInfo());
+    }
+
+    protected void testStudentInfoTable(StudentInfoTable studentInfoTable) {
+        assertNotNull(studentInfoTable);
+        
+        assertEquals(inputStudent.name, studentInfoTable.getName());
+        assertEquals(inputStudent.email, studentInfoTable.getEmail());
+        assertEquals(inputStudent.section, studentInfoTable.getSection());
+        assertEquals(inputStudent.team, studentInfoTable.getTeam());
+        assertEquals(inputStudent.comments, studentInfoTable.getComments());
+        assertEquals(inputStudent.course, studentInfoTable.getCourse());
+        assertEquals(isAbleToAddComment, studentInfoTable.isAbleToAddComment());
+        assertEquals(hasSection, studentInfoTable.getHasSection());
+    }
+    
+    private void createStudentData(String pictureKey) {
+        String name = "John Doe";
+        String email = "john@doe.com";
+        
+        createStudent(name, email);
+        createStudentProfile(name, email, pictureKey);
+    }
+    
+    protected void createStudent(String name, String email) {
+        String comments = "This is a comment for John Doe.";
+        String courseId = "CourseForJohnDoe";
+        String team = "TeamForJohnDoe";
+        String section = "SectionForJohnDoe";
+        
+        inputStudent = new StudentAttributes(null, email, name, comments, courseId, team, section);
+    }
+    
+    private void createStudentProfile(String name, String email, String pictureKey) {
+        String shortName = "John";
+        String institute = "InstituteForJohnDoe";
+        String nationality = "NationForJohnDoe";
+        String gender = Const.GenderTypes.MALE;
+        String moreInfo = "Information for John Doe.";
+        
+        if (pictureKey == null || pictureKey.isEmpty()) {
+            this.pictureUrl = Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH;
+        } else {
+            this.pictureUrl =  Const.ActionURIs.STUDENT_PROFILE_PICTURE + "?"
+                               + Const.ParamsNames.BLOB_KEY + "=" + pictureKey + "&"
+                               + Const.ParamsNames.USER_ID + "=null";
+        }
+        
+        inputStudentProfile = new StudentProfileAttributes(
+                null, shortName, email, institute, nationality, gender, moreInfo, pictureKey);
+    }
+    
+    private void initCommonData() {
```

Comment:
Hmm, it's called several times and not really "initialising", I'll give it a more accurate name instead.